src/protocol/frame/frame.rs

```diff
@@ -203,19 +206,55 @@ impl FrameHeader {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+enum Payload {
```

Comment:
Not sure that follow, but  can be this new type be replaced by 

[Bytes::is_unique](https://docs.rs/bytes/latest/bytes/struct.Bytes.html#method.is_unique) and `Bytes::to_vec` ?

If no clone of `Bytes` create from `Vec`, than `is_unique` should gives `true`.

Reply:
Do you mean that instead of a separate `Payload` type, we could use `Bytes`? (note: this is a payload of a frame, not the `Message`; the `Message` uses `Bytes`).

Using `Bytes` as a type of the `payload` of a `Frame` would be possible if not masking that we need to perform when e.g. sending frames from the client. `Bytes` is immutable. `BytesMut` is, though. But converting it to `BytesMut` upfront (before we know that we'll mutate the payload by applying a mask) is probably not optimal (I hope I understood your question correctly).